File: src/types/product.ts
export type ProductCategory =
  | 'smartphones'
  | 'laptops'
  | 'audio'
  | 'gaming'
  | 'home'
  | 'accessories';

export type ProductItem = {
  id: string;
  title: string;
  subtitle: string;
  category: ProductCategory;
  brand: string;
  price: number;
  previousPrice?: number;
  badge?: string;
  rating: number;
  reviewsCount: number;
};

export type ProductSpecification = {
  label: string;
  value: string;
};

export type ProductDetails = {
  overview: string;
  highlights: string[];
  specifications: ProductSpecification[];
  included: string[];
};

export type CatalogCategory = {
  id: ProductCategory;
  title: string;
  description: string;
};

export type CatalogProductRecord = ProductItem & {
  details: ProductDetails;
};

export type CatalogDatabase = {
  version: number;
  seededAt: string;
  categories: CatalogCategory[];
  products: CatalogProductRecord[];
};

export type ProductPriceFilter =
  | 'all'
  | 'under-10000'
  | '10000-30000'
  | '30000-plus'
  | 'discount-only';

File: backend/Rozetka.Api/Models/Product.cs
namespace Rozetka.Api.Models;

public class Product
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Sku { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Subtitle { get; set; } = string.Empty;
    public string Brand { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public decimal? PreviousPrice { get; set; }
    public string Badge { get; set; } = string.Empty;
    public double Rating { get; set; }
    public int ReviewsCount { get; set; }
    public string ImageUrl { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string ManufacturerUrl { get; set; } = string.Empty;
    public string Specifications { get; set; } = string.Empty;
    public int StockQuantity { get; set; }
    public Guid CategoryId { get; set; }
    public Category? Category { get; set; }
    public ICollection<Review> Reviews { get; set; } = new List<Review>();
}
